mlflow/utils/__init__.py

```diff
@@ -6,12 +6,15 @@
 from itertools import islice
 from sys import version_info
 
-_logger = logging.getLogger(__name__)
-
+import pydantic
+from packaging.version import Version
 
 PYTHON_VERSION = f"{version_info.major}.{version_info.minor}.{version_info.micro}"
 
 
+IS_PYDANTIC_V2 = Version(pydantic.__version__).major >= 2
```

Comment:
```suggestion
IS_PYDANTIC_V2_OR_NEWER = Version(pydantic.__version__).major >= 2
```

Let's be more precise here :)